packages/core/src/common/application-error.ts

```diff
@@ -32,11 +32,7 @@ export namespace ApplicationError {
         code: C;
         is(arg: object | undefined): arg is ApplicationError<C, D>
     }
-    const codes: number[] = [];
     export function declare<C extends number, D>(code: C, factory: (...args: any[]) => Literal<D>): Constructor<C, D> {
-        if (codes.indexOf(code) !== -1) {
```

Comment:
Not sure about it. It makes defining error codes per a service easier, since you don't need to care about errors from other services. But if a client using several services it could lead to issues.

Reply:
reverted